src/components/progress-circle/progress-circle.ts

```diff
@@ -96,11 +191,93 @@ export class MdProgressCircle {
   },
   templateUrl: './components/progress-circle/progress-circle.html',
   styleUrls: ['./components/progress-circle/progress-circle.css'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class MdSpinner extends MdProgressCircle {
-  constructor() {
-    super();
+  constructor(changeDetectorRef: ChangeDetectorRef) {
+    super(changeDetectorRef);
     this.mode = 'indeterminate';
   }
 }
+
+
+/**
+ * Module functions.
+ */
+
+/** Clamps a value to be between 0 and 100. */
+function clamp(v: number) {
+  return Math.max(0, Math.min(100, v));
+}
+
+
+/**
+ * Returns the current timestamp either based on the performance global or a date object.
+ */
+function now() {
+  if (typeof performance !== 'undefined') {
+    return performance.now();
+  }
+  return Date.now();
+}
+
+
+/**
+ * Converts Polar coordinates to Cartesian.
+ */
+function polarToCartesian(radius: number, pathRadius: number, angleInDegrees: number) {
+  var angleInRadians = (angleInDegrees - 90) * DEGREE_IN_RADIANS;
+
+  return (radius + (pathRadius * Math.cos(angleInRadians))) +
+    ',' + (radius + (pathRadius * Math.sin(angleInRadians)));
+}
+
+
+/**
+ * Easing function for linear animation.
+ */
+function linearEase(currentTime: number, startValue: number,
+                    changeInValue: number, duration: number) {
+  return changeInValue * currentTime / duration + startValue;
+}
+
+
+/**
+ * Easing function to match material design indeterminate animation.
+ */
+function materialEase(currentTime: number, startValue: number,
+                      changeInValue: number, duration: number) {
+  var ts = (currentTime /= duration) * currentTime;
+  var tc = ts * currentTime;
+  return startValue + changeInValue * (6 * tc * ts + -15 * ts * ts + 10 * tc);
+}
+
+
+/**
+ * Determines the path value to define the arc.
+ */
+function getSvgArc(current: number, rotation: number) {
+  // The angle can't be exactly 360, because the arc becomes hidden.
+  var maximumAngle = 359.99 / 100;
+  var startPoint = rotation || 0;
+  var radius = 50;
+  var pathRadius = 40;
+
+  var startAngle = startPoint * maximumAngle;
+  var endAngle = current * maximumAngle;
+  var start = polarToCartesian(radius, pathRadius, startAngle);
+  var end = polarToCartesian(radius, pathRadius, endAngle + startAngle);
+  var arcSweep = endAngle < 0 ? 0 : 1;
+  var largeArcFlag: number;
+
+  if (endAngle < 0) {
+    largeArcFlag = endAngle >= -180 ? 0 : 1;
+  } else {
+    largeArcFlag = endAngle <= 180 ? 0 : 1;
+  }
+
+  return `M${start}A${pathRadius},${pathRadius} 0 ${largeArcFlag},${arcSweep} ${end}`;
+}
+
+function pageVisible() {
+  return !document.hidden;
```

Comment:
I'd call this function `isPageVisible`, but then you might as well just use `document.hidden` directly since it's not much shorter. Up to you.